src/test/ui/suggestions/issue-104287.rs

```diff
@@ -1,9 +1,13 @@
 // The purpose of this test is not to validate the output of the compiler.
 // Instead, it ensures the suggestion is generated without performing an arithmetic overflow.
 
+struct S;
```

Comment:
Why did you change this UI test? I think the UI test was added specifically to check for an ICE that resulted specifically from this combination of resolver errors.

Reply:
Because the error that produced the ICE would no longer emit with the correct propagation of `[type error]`. The relevant part seemed to be the error that I kept.

Reply:
Hm, I'm not sure if this modified UI test is "faithful" to the ICE anymore, either, though. I'm tempted to leave it as-is even if the error output turns into basically nothing. Either that, or we should just yeet this test completely.. 

I guess I don't care strongly any way.